plugin/trino-hive/src/main/java/io/trino/plugin/hive/HiveStorageFormat.java

```diff
@@ -148,6 +149,15 @@ public DataSize getEstimatedWriterMemoryUsage()
         return estimatedWriterMemoryUsage;
     }
 
+    public boolean isSplittable(String path)
+    {
+        // Only uncompressed text input format is splittable
+        return switch (this) {
+            case ORC, PARQUET, AVRO, RCBINARY, RCTEXT, SEQUENCEFILE -> true;
+            case JSON, OPENX_JSON, TEXTFILE, CSV, REGEX -> CompressionKind.forFile(path).isEmpty();
```

Comment:
Maybe add a variable to the enum like "isTextBasedInputFormat"

Reply:
I thought about that, but a bunch of true/false flags seems harder to read than this explicit list.